integrations/nvidia/src/haystack_integrations/components/generators/nvidia/generator.py

```diff
@@ -0,0 +1,162 @@
+
+        self._client = NvidiaCloudFunctionsClient(
+            api_key=api_key,
+            headers={
+                "Content-Type": "application/json",
+                "Accept": "application/json",
+            },
+        )
+
+        if self._model not in SUPPORTED_MODELS:
+            models = ", ".join(e.value for e in NvidiaGeneratorModel)
+            msg = f"Model {self._model} is not supported, available models are: {models}"
+            raise ValueError(msg)
+
+    def warm_up(self):
+        """
+        Initializes the component.
+        """
+        if self._model_id is not None:
+            return
+        self._model_id = self._client.get_model_nvcf_id(str(self._model))
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns:
+            Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self, model=str(self._model), api_key=self._api_key.to_dict(), model_arguments=self._model_arguments
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "NvidiaGenerator":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data:
+            Dictionary to deserialize from.
+        :returns:
+           Deserialized component.
+        """
+        init_params = data.get("init_parameters", {})
+        deserialize_secrets_inplace(init_params, ["api_key"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(replies=List[str], meta=List[Dict[str, Any]])
+    def run(self, prompt: str):
+        """
+        Queries the model with the provided prompt.
+
+        :param prompt:
+            Text to be sent to the generative model.
+        :returns:
+            A dictionary with the following keys:
+            - "replies": replies generated by the model.
+            - "meta": metadata for each reply.
+        """
+        if self._model_id is None:
+            msg = "The generation model has not been loaded. Call warm_up() before running."
+            raise RuntimeError(msg)
+
+        messages = [Message(role="user", content=prompt)]
+        request = GenerationRequest(messages=messages, **self._model_arguments).to_dict()
+        json_response = self._client.query_function(self._model_id, request)
+
+        replies = []
+        meta = []
+        data = GenerationResponse.from_dict(json_response)
+        for choice in data.choices:
+            replies.append(choice.message.content)
+            meta.append(
+                {
+                    "role": choice.message.role,
+                    "finish_reason": choice.finish_reason,
+                    # The usage field is not part of each choice, so we use reuse it each time
+                    "completion_tokens": data.usage.completion_tokens,
+                    "prompt_tokens": data.usage.prompt_tokens,
+                    "total_tokens": data.usage.total_tokens,
```

Comment:
Let's just provide these as a separate `usage` key in the outputs (as their show the cumulative usage).